### ISSUE TITLE
Payments Past Due System

### ISSUE THREAD

**Issue by m-henderson** [mentorient/mentorient#16]
We need a way for the user of the application to be able to edit the tenant and set them past due if they are past due on their rent. Also, for future reference, we need to be thinking of an alert system that will email the user of the application if their tenant is past due.

**Comment by me245** [comment]
@m-henderson How full featured were you envisioning this system? I see the accounting tab on the application, so were you hoping for a system that keeps track of payments made with the user inputing the values into a form to keep track? Or were you hoping to build out an invoicing and potentially a payment system?

For right now I am going to assume that there will be at least a payment tracking system. If that is correct, I would be inclined to put in some automation. IE tenant is configured to have payments and a payment amount by a certain time. If that hasn't been entered by the due_by date then they are automatically marked late. What are your thoughts on that sort of workflow? It would allow for further automation of building out an invoicing system and then even a payments system afterwards.

**Comment by m-henderson** [comment]
@me245 first of all, I would like to start out by saying thank you for your contributions. Secondly, I do not feel the need, at this time, for a system that processes payments. I have spoken with a few people in real estate and they deal mostly with cash. They just want a way to track, and change as needed, who has paid, who is past due, what the tenant's monthly payment is, etc. Then, they want to see what their projected monthly income is based on that data.

**Comment by me245** [comment]
Gotcha. So do you want me to give you a pull request example of what I am thinking about as far as a receipt tracking system would be like? I am thinking, in terms of a user story it would look like this:

As a landlord 
  I want to be able to log in and see which tenants are late
  So that I can follow up with them via face2face, phone/text, email
Acceptance Criteria:
Criteria1:
   Given I have entered in default payment information or customized that information for a specific 
             payment
   When I have not entered in payment information for a payment that has a payment due field in the 
              past
   Then I am notified by some visual indication on the website or via email(future plan)

And if I was super thorough I would make similar stories for all of the dependent actions in the criteria, but I think it gets the point across as of right now. Does that work for you?

**Comment by me245** [comment]
As a note for payment systems, if that is a feature that is requested I would suggest offloading a fair chunk of that to a 3rd party solution so that we don't have to worry to the same extent for compliance issues. It could integrate with https://stripe.com/ for example

**Comment by m-henderson** [comment]
@me245 I agree! I think if we can put the liability on a 3rd party, then this will be a great feature!
